src/libraries/CallLib.sol

```diff
@@ -31,7 +31,7 @@ library CallLib {
                 }
             }
 
-            return keccak256(packedHashes);
+            return keccak256(abi.encodePacked(packedHashes, nonce));
```

Comment:
Ah I don't think the nonce should be hashed with the calls here, but rather in MinimalDelegation where we build the digest we want to verify against. This library should be strictly for operations on Call

Reply:
yeah this would break EIP712 spec actually because it needs to match a typehash

would recommend defining an outermost struct that holds the array of calls and nonce

`Execute(Call[] calls, uint256 nonce)`

Reply:
yea agree on the outermost struct

Reply:
oh so u mean adding another struct that contains an array of `Calls` and a nonce? that way it doesn't break EIP712? should this still be in the Call library?

Reply:
this would probably live in a different library since it has its own typehash, etc.